File: frontend/src/app/models/product.ts
export interface Product {
  id: number;
  productName: string;
  productCode: string;
  pricePerPiece: number;
  quantityInStock: number;
}

export interface StockItem extends Product {
  qty: number;
}

export interface InsufficientItem {
  productId: number;
  availableQty: number;
}

File: backend/Domain/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace Infrastructure;

public partial class Product
{
    public int Id { get; set; }

    public string ProductName { get; set; } = null!;

    public double PricePerPiece { get; set; }

    public string? ProductCode { get; set; }
}
